### PR TITLE
osd/scheduler/mClockScheduler: Use same profile and client ids for all clients to ensure allocated QoS limit consumption.

### PR DESCRIPTION
**Pull request by sseshasa**
Currently, without the distributed mClock (dmClock) feature, all external clients classified under op_scheduler_class::client must share the allocated reservation and limit as set within default_external_client_info. This is realized only if all the external clients are assigned the same id. This was not true because the client_id field within the client_profile_id_t was set to item.get_owner() which uniquely identified each client. As a result, an increase in the client count resulted in a proportional increase in the consumption of the allocated limit which is not the desired outcome in the current implementation.

The fix is to use the same client_id and profile_id within the client_profile_id_t structure for all clients. These fields are set to '0'. This ensures that all such clients get the same QoS controls which means reservation and limit is shared between them. In other words, regardless of the number of clients active, the reservation and limit consumption remains constant as per the mClock profile setting and is shared amongst them.

NOTE: When dmClock feature (client vs client QoS) is implemented, the above fields would change with client_id set to the global id (client.####) and the profile_id set to the id generated by the client's QoS profile.

Tests were performed using rados bench with multiple instances on a single OSD to confirm that the limit allocation is
not breached. Test results will be updated soon.

Fixes: https://tracker.ceph.com/issues/62293
Signed-off-by: Sridhar Seshasayee <sseshasa@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by sseshasa** [comment]
**Test Results**
Random write tests were performed on a single OSD  with 5 Rados bench instances.

**Test type**: Random write
**Block Size**: 4 KiB
**Object Size**: 4 KiB
**Concurrent Writes**: 16
**Number of OSDs**: 1 [pool size = 1]

mClock QoS_Limited_Bandwidth_Sharing_HDD

mClock QoS_Limited_Bandwidth_Sharing_SSD

**Comment by jianwei1216** [comment]
@sseshasa  look good to me, thanks share

- background_recovery  client:{ class_id: 0 client_id: 0 profile_id: 0 }
- radosclient                    client:{ class_id: 3 client_id: 0 profile_id: 0 }
```
# date; ceph daemon osd.0 dump_op_pq_state
Mon Aug  7 05:04:21 UTC 2023
{
    "OSD:ShardedOpWQ:0": {
        "queue_sizes": {
            "immediate": 0,
            "scheduler": 9
        },
        "mClockClients": {
            "client_count": 2,
            "clients": "{ 
                 PriorityQueue:: 
                     { client:{ class_id: 0 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } }  
                     { client:{ class_id: 3 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } req_count:7 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } 
                     { reservation_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false  r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } 
                     { ready_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } req_count:7 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } 
                     { limit_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } }"
        },
        "mClockQueues": {
            "queues": "
                   RESER:
                       { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, 
                       { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } req_count:7 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }
                   LIMIT:
                       { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, 
                      { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } req_count:7 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }
                   READY:
                       { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } req_count:7 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0223 p:384655.0603 l:384663.0223 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }, 
                       { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1118 p:384736.4122 l:384662.1525 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }"
        }
    },
    "OSD:ShardedOpWQ:1": {
        "queue_sizes": {
            "immediate": 0,
            "scheduler": 18
        },
        "mClockClients": {
            "client_count": 2,
            "clients": "{ PriorityQueue::  { client:{ class_id: 0 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } }  { client:{ class_id: 3 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { reservation_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } { ready_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { limit_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } }"
        },
        "mClockQueues": {
            "queues": "RESER:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }LIMIT:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }READY:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0292 p:384659.0230 l:384663.0292 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } req_count:9 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0930 p:384741.4377 l:384662.1527 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }"
        }
    },
    "OSD:ShardedOpWQ:2": {
        "queue_sizes": {
            "immediate": 0,
            "scheduler": 8
        },
        "mClockClients": {
            "client_count": 2,
            "clients": "{ PriorityQueue::  { client:{ class_id: 0 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } }  { client:{ class_id: 3 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } req_count:3 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { reservation_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } { ready_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } req_count:3 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { limit_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } }"
        },
        "mClockQueues": {
            "queues": "RESER:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } req_count:3 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }LIMIT:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } req_count:3 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }READY:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } req_count:3 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0406 p:384656.0633 l:384663.0406 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } req_count:5 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.1851 p:384735.0865 l:384662.1628 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }"
        }
    },
    "OSD:ShardedOpWQ:3": {
        "queue_sizes": {
            "immediate": 0,
            "scheduler": 3
        },
        "mClockClients": {
            "client_count": 2,
            "clients": "{ PriorityQueue::  { client:{ class_id: 0 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } }  { client:{ class_id: 3 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } req_count:1 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { reservation_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } { ready_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } req_count:1 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } } } { limit_top:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } } } }"
        },
        "mClockQueues": {
            "queues": "RESER:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } req_count:1 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }LIMIT:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } req_count:1 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }READY:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } req_count:1 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384663.0412 p:384662.0562 l:384663.0412 } client:{ class_id: 3 client_id: 0 profile_id: 0 } } }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } req_count:2 top_req:{ ClientReq:: tag:{ RequestTag:: ready:false r:384662.0965 p:384726.7455 l:384662.0965 } client:{ class_id: 0 client_id: 0 profile_id: 0 } } }"
        }
    },
    "OSD:ShardedOpWQ:4": {
        "queue_sizes": {
            "immediate": 0,
            "scheduler": 0
        },
        "mClockClients": {
            "client_count": 2,
            "clients": "{ PriorityQueue::  { client:{ class_id: 0 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384661.9229 p:384730.2963 l:384661.9628 } req_count:0 top_req:none } }  { client:{ class_id: 3 client_id: 0 profile_id: 0 }, record:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none } } { reservation_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none } } { ready_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none } } { limit_top:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none } } }"
        },
        "mClockQueues": {
            "queues": "RESER:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384661.9229 p:384730.2963 l:384661.9628 } req_count:0 top_req:none }LIMIT:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384661.9229 p:384730.2963 l:384661.9628 } req_count:0 top_req:none }READY:{ ClientRec:: client:{ class_id: 3 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384662.0295 p:384659.0593 l:384662.0295 } req_count:0 top_req:none }, { ClientRec:: client:{ class_id: 0 client_id: 0 profile_id: 0 } prev_tag:{ RequestTag:: ready:false r:384661.9229 p:384730.2963 l:384661.9628 } req_count:0 top_req:none }"
        }
    }
}

```

**Comment by sseshasa** [comment]
> @sseshasa look good to me, thanks share
> 
>     * background_recovery  client:{ class_id: 0 client_id: 0 profile_id: 0 }
> 
>     * radosclient                    client:{ class_id: 3 client_id: 0 profile_id: 0 }
> 
@jianwei1216  Thanks for bringing up this issue and your feedback.

File: src/osd/scheduler/mClockScheduler.h [hunk 1]
```diff
@@ -38,12 +38,36 @@ constexpr double default_max = std::numeric_limits<double>::is_iec559 ?
...
+ * bandwidth.
+ *
+ * Note: Post Reef, both members will be set to non-zero
+ * values when the distributed feature of the mClock
+ * algorithm is utilized.
+ */
 struct client_profile_id_t {
-  client_id_t client_id;
-  profile_id_t profile_id;
+  uint64_t client_id;
```
Comment (athanatos) [inline src/osd/scheduler/mClockScheduler.h:57]: = 0

File: src/osd/scheduler/mClockScheduler.h [hunk 2]
```diff
@@ -38,12 +38,36 @@ constexpr double default_max = std::numeric_limits<double>::is_iec559 ?
...
+  uint64_t profile_id;
```
Comment (athanatos) [inline src/osd/scheduler/mClockScheduler.h:58]: = 0

File: src/osd/scheduler/mClockScheduler.h [hunk 3]
```diff
@@ -38,12 +38,36 @@ constexpr double default_max = std::numeric_limits<double>::is_iec559 ?
...
+
+  client_profile_id_t(uint64_t _client_id, uint64_t _profile_id) :
+    client_id(_client_id),
+    profile_id(_profile_id)
```
Comment (athanatos) [inline src/osd/scheduler/mClockScheduler.h:62]: You don't need a comment indicating that the body is empty: ```    profile_id(_profile_id) {}```

File: src/osd/scheduler/mClockScheduler.h [hunk 4]
```diff
@@ -38,12 +38,36 @@ constexpr double default_max = std::numeric_limits<double>::is_iec559 ?
...
+  {
+    // empty
+  }
+
+  client_profile_id_t() : client_profile_id_t(0, 0)
```
Comment (athanatos) [inline src/osd/scheduler/mClockScheduler.h:67]: client_profile_id_t() = default;

File: src/osd/scheduler/mClockScheduler.h (hunk 1 above)
Comment (sseshasa) [reply src/osd/scheduler/mClockScheduler.h]: Done.

File: src/osd/scheduler/mClockScheduler.h (hunk 2 above)
Comment (sseshasa) [reply src/osd/scheduler/mClockScheduler.h]: Done.

File: src/osd/scheduler/mClockScheduler.h (hunk 3 above)
Comment (sseshasa) [reply src/osd/scheduler/mClockScheduler.h]: Done.

File: src/osd/scheduler/mClockScheduler.h (hunk 4 above)
Comment (sseshasa) [reply src/osd/scheduler/mClockScheduler.h]: Done.

**Review by athanatos — APPROVED** [review]

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcMEs20HAJ1828-wip-yuri11-testing-2023-08-17-0823

**Pull request merged**